core/include/gnuradio-4.0/CircularBuffer.hpp

```diff
@@ -285,118 +285,121 @@ class CircularBuffer
+                    // mirror samples below/above the buffer's wrap-around point
+                    const size_t nFirstHalf = std::min(size - _parent->_index, _parent->nSamplesPublished());
+                    const size_t nSecondHalf = _parent->nSamplesPublished() - nFirstHalf;
 
-        if constexpr (isMultiThreadedStrategy()) {
-            if (_parent->_rangesCounter == 0 && !isFullyPublished()) {
-                fmt::print(stderr, "CircularBuffer::multiple_writer::PublishableOutputRange() - did not publish {} samples\n", _parent->_internalSpan.size() - _parent->_nSamplesPublished);
-                std::abort();
-            }
+                    auto &data = _parent->_buffer->_data;
+                    std::copy(&data[_parent->_index], &data[_parent->_index + nFirstHalf], &data[_parent->_index + size]);
+                    std::copy(&data[size], &data[size + nSecondHalf], &data[0]);
+                }
+                _parent->_claimStrategy->publish(_parent->_offset, _parent->nSamplesPublished());
+                _parent->_offset += static_cast<signed_index_type>(_parent->nSamplesPublished());
+#ifndef NDEBUG
+                if constexpr (isMultiThreadedStrategy()) {
+                    if (!isFullyPublished()) {
+                        fmt::print(stderr, "CircularBuffer::multiple_writer::PublishableOutputRange() - did not publish {} samples\n", _parent->_internalSpan.size() - _parent->_nSamplesPublished);
+                        std::abort();
+                    }
 
-        } else {
-            if (_parent->_rangesCounter == 0 && !_parent->_internalSpan.empty() && !isPublished()) {
-                fmt::print(stderr, "CircularBuffer::single_writer::PublishableOutputRange() - omitted publish call for {} reserved samples\n", _parent->_internalSpan.size());
-                std::abort();
-            }
+                } else {
+                    if (!_parent->_internalSpan.empty() && !isPublished()) {
+                        fmt::print(stderr, "CircularBuffer::single_writer::PublishableOutputRange() - omitted publish call for {} reserved samples\n", _parent->_internalSpan.size());
+                        std::abort();
+                    }
+                }
+#endif
+                _parent->_nSamplesPublished = 0;
+                _parent->_internalSpan = {};
+            };
         }
-    #endif
-    }
 
-    [[nodiscard]] constexpr bool
-    isPublished() const noexcept {
-        return _parent->_isRangePublished;
-    }
+        [[nodiscard]] constexpr std::size_t
+        samplesToPublish() const noexcept {
+            return _parent->_nSamplesPublished;
+        }
 
-    [[nodiscard]] constexpr bool
-    isFullyPublished() const noexcept {
-        return _parent->_internalSpan.size() == _parent->_nSamplesPublished;
-    }
+        [[nodiscard]] constexpr bool
+        isPublished() const noexcept {
+            return _parent->_isRangePublished;
+        }
```

Comment:
It looks like you are using wrong clang-format. The return type must be on the same line. This appear in several places. Please check.